apps/vaporgui/VDoubleLineEdit.cpp

```diff
@@ -0,0 +1,79 @@
+#include <string>
+#include <sstream>
+#include <iomanip>
+
+#include <QString>
+
+#include "VDoubleLineEdit.h"
+
+VDoubleLineEdit::VDoubleLineEdit( double value, bool useMenu ) : 
+    VNumericLineEdit(),
+    _value( value )
+{
+    // Disconnect the QLineEdit's interaction with VStringLineEdit::_valueChanged(),
+    // and reconnect it to VDoubleLineEdit::_valueChanged()
+    disconnect( _lineEdit, SIGNAL( editingFinished() ),
+        this, SLOT( _valueChanged() ) );
+    connect( _lineEdit, SIGNAL( editingFinished() ),
```

Comment:
Why do you disconnect the original disconnect and reconnect it to the same slot? (In VDoubleLineEdit and VIntLineEdit constructors). Shouldn’t this be done with inheritance? Another option would be to connect VStringEdit::changed to VIntEdit::changed. Also, VStringEdit::changed ensures that the changed signal is not emitted if the user re-enters the same value. This functionality is lost for the numeric edits in the current implementation because of the above.

Reply:
Thanks.  I incorrectly assumed that the parent's implementation of _valueChanged() would hold the signal slot connection.